### PR TITLE
Add support for user activation using activation_token

### PR DESCRIPTION
**Pull request by arvindkrishnakumar-okta**
OKTA-470731

### CODE REVIEW

File: api/src/main/java/com/okta/idx/sdk/api/model/TokenType.java [hunk 1]
```diff
@@ -19,7 +19,11 @@
 
     ACCESS_TOKEN("access_token"),
 
-    REFRESH_TOKEN("refresh_token");
+    REFRESH_TOKEN("refresh_token"),
+
+    ACTIVATION_TOKEN("activation_token"),
```
Comment (JayNewstrom) [inline api/src/main/java/com/okta/idx/sdk/api/model/TokenType.java:24]: This seems like it should be another enum type. It doesn’t really mesh with access token/refresh token.

File: api/src/test/groovy/com/okta/idx/sdk/api/client/IDXAuthenticationWrapperTest.groovy [hunk 2]
```diff
@@ -355,6 +355,38 @@ class IDXAuthenticationWrapperTest {
         assertThat(authenticationResponse.getTokenResponse().getIdToken(), notNullValue())
     }
 
+    @Test
+    void beginTransactionWithAndWithoutActivationTokenTest() {
```
Comment (JayNewstrom) [inline api/src/test/groovy/com/okta/idx/sdk/api/client/IDXAuthenticationWrapperTest.groovy:359]: Is this name right? It says without activation token. But it seems like it’s with activation token.

**Review by JayNewstrom — APPROVED** [review]
I left a few comments that I think are worth considering, but nothing that’s a show stopper.

File: api/src/test/groovy/com/okta/idx/sdk/api/client/IDXAuthenticationWrapperTest.groovy (hunk 2 above)
Comment (arvindkrishnakumar-okta) [reply api/src/test/groovy/com/okta/idx/sdk/api/client/IDXAuthenticationWrapperTest.groovy:359]: yes, it tests both [with](https://github.com/okta/okta-idx-java/blob/rc_2.0.0/api/src/test/groovy/com/okta/idx/sdk/api/client/IDXAuthenticationWrapperTest.groovy#L372) and [without](https://github.com/okta/okta-idx-java/blob/rc_2.0.0/api/src/test/groovy/com/okta/idx/sdk/api/client/IDXAuthenticationWrapperTest.groovy#L383) activation token.

File: api/src/main/java/com/okta/idx/sdk/api/model/TokenType.java (hunk 1 above)
Comment (arvindkrishnakumar-okta) [reply api/src/main/java/com/okta/idx/sdk/api/model/TokenType.java:24]: sure, will move it to a new enum type.

Comment (arvindkrishnakumar-okta) [reply api/src/main/java/com/okta/idx/sdk/api/model/TokenType.java]: done, updated PR.

**Comment by arvindkrishnakumar-okta** [comment]
> I left a few comments that I think are worth considering, but nothing that’s a show stopper.

sure, addressed both of them.

**Review by JayNewstrom — APPROVED** [review]

**Pull request merged**